moduleframework/dockerlinter.py

```diff
@@ -161,11 +166,72 @@ def get_specific_label(self, label_name=None):
         if label_name is None:
             return []
         label_list = self.get_docker_labels()
-        return [label_name in label_list]
+        return [label_list[key] for key in label_list.keys() if label_name == key]
 
-    def check_baseruntime(self):
+    def check_from_is_first(self):
         """
-        Function returns docker labels
-        :return: label dictionary
+        Function checks if FROM directive is really first directive.
+        :return: True if FROM is first, False if FROM is not first directive
+        """
+        if self.dfp_structure[0].get('instruction') == 'FROM':
+            return True
+        else:
+            return False
+
+    def check_from_directive_is_valid(self):
+        """
+        Function checks if FROM directive contains valid format like is specified here
+        http://docs.projectatomic.io/container-best-practices/#_line_rule_section
+        Regular expression is: ^[a-z0-9.]+(\/[a-z0-9\D.]+)+$
+        Example registry:
+            registry.fedoraproject.org/f26/etcd
+            registry.fedoraproject.org/f26/flannel
+            registry.access.redhat.com/rhscl/nginx-18-rhel7
+            registry.access.redhat.com/rhel7/rhel-tools
+            registry.access.redhat.com/rhscl/postgresql-95-rhel7
+
+        :return:
+        """
+        correct_format = False
+        struct = self.dfp_structure[0]
+        if struct.get(INSTRUCT) == 'FROM':
+            p = re.compile("^[a-z0-9.]+(\/[a-z0-9\D.]+)+$")
+            if p.search(struct.get('value')) is not None:
+                correct_format = True
+        return correct_format
+
+    def check_chained_run_command(self):
+        """
+        Function checks if Dockerfile does not contain more RUN commands in two or more rows.
+        BAD examples:
+             FROM fedora
+             RUN ls /
+             RUN cd /
+        GOOD example:
+             FROM fedora
+             RUN ls / && cd /
+        :return: True if Dockerfile does not contain RUN instructions in rows
+                False if Dockerfile contains more RUN instructions in rows
+        """
+        value = 0
+        for struct in self.dfp_structure:
+            if struct.get(INSTRUCT) == RUN:
+                value += 1
+        if int(value) > 1:
```

Comment:
this seems to permit only a single `RUN` instruction, that's not good

Reply:
Well e.g.

~~~
FROM fedora
RUN dnf install foobar1 && dnf clean all
RUN ls /
RUN cd /
~~~
This should be just two RUN instructions.

~~~
FROM fedora
RUN dnf install foobar1
RUN dnf install clean all
RUN ls /
RUN cd /
~~~

This should be also two RUN instructions.

I would prefer again separate it into two tests. One for testing `dnf` and stuff around it and second one for testing rest instructions.

WDYT? I hate sort cuts.

Reply:
Sounds like a good idea since you correctly pointed out those are 2 separate use cases.

Bear in mind that current implementation just counts `RUN`s, so e.g. [this dockerfile](https://github.com/container-images/postfix/blob/master/Dockerfile) would not pass.